File: go/command_injection/file_operations.go
package commandinjection

import (
	"fmt"
	"os/exec"
)

// VULN 1: exec.Command sh -c with string concatenation - file conversion
func ConvertFileFormat(inputFile, outputFormat string) ([]byte, error) {
	cmd := exec.Command("sh", "-c", "convert "+inputFile+" output."+outputFormat)
	return cmd.Output()
}

// VULN 2: exec.Command sh -c with fmt.Sprintf - create zip archive
func CreateArchive(directory, archiveName string) ([]byte, error) {
	cmdStr := fmt.Sprintf("zip -r %s.zip %s", archiveName, directory)
	cmd := exec.Command("sh", "-c", cmdStr)
	return cmd.Output()
}

// VULN 3: exec.Command sh -c with string concatenation - chmod
func SetFilePermissions(filepath, permissions string) ([]byte, error) {
	cmd := exec.Command("sh", "-c", "chmod "+permissions+" "+filepath)
	return cmd.Output()
}

File: python/command_injection/file_operations.py
import os
import subprocess


# VULN 1: os.system with string concatenation - file format conversion
def convert_file_format(input_file, output_format):
    os.system("convert " + input_file + " output." + output_format)


# VULN 2: subprocess.call with shell=True - create zip archive
def create_archive(directory, archive_name):
    subprocess.call("zip -r " + archive_name + ".zip " + directory, shell=True)


# VULN 3: os.popen with f-string - set file permissions
def set_file_permissions(filepath, permissions):
    result = os.popen(f"chmod {permissions} {filepath}")
    return result.read()
